### ISSUE TITLE
Using array notation breaks when values has periods

### ISSUE THREAD

**Issue by blakeembrey** [jonschlinkert/get-value#7]
Converting an array to a string by naive join results in periods that were previously in a key becoming two keys.

It'd be nicer if array notation was just passed through ad-hoc, and instead the string syntax is converted to array notation - the current method is kind of jank, array to string back to array with changes each step.

**Comment by blakeembrey** [comment]
@jonschlinkert Would you accept a PR tidying up the API surface?

**Comment by jonschlinkert** [comment]
> Converting an array to a string by naive join results in periods that were previously in a key becoming two keys.

Technically it's an _explicit_ join https://github.com/jonschlinkert/get-value#usage. Dots that should not be split should be escaped. In practical use, we've tried many ways and this is the only way that consistently produces what the user expects. 

Of course, I'd be happy to accept a pr for doing this differently. But the code was designed to normalize object paths. 

We do this quite often:

```js
app.set(['cache.data', key], val);
```

Where `key` might be `foo` or `a.b.c`, etc.  This pattern cuts down on a lot of logic that would otherwise be needed to accomplish the same thing. 

That said, you write outstanding code, I would always take a pr from you to to improve the code. whatever suggestions you have I'd love to hear

**Comment by jonschlinkert** [comment]
fwiw I can also see why it would be nice to not have to escape keys when using a string-array combination. e.g. that combination seems like a good way of avoiding the need to escape. 

I wouldn't be opposed to the idea of exposing a method for your use case. thoughts?

**Comment by blakeembrey** [comment]
Ah, interesting - it's not explicitly in the docs, I assumed that was only intended for the string implementation. Perhaps there could be a method to parse the string exposed, then you could do string or `set.split('cache.data').concat(key)`? Internally if you pass only a string it could use the same.

FWIW, the reason I run into this was I'm actually storing URIs as the keys (https://github.com/blakeembrey/node-scrappy/blob/master/src/extract.ts#L110) and it confused me for a while to find that the data wasn't being selected properly. Ended up implementing my own, since it's simple, but I'd like to normalise this module for the next guy.

**Comment by jonschlinkert** [comment]
> it's not explicitly in the docs

Thanks for pointing that out! when I looked at the readme earlier, I saw the dot-escaping section and didn't even think about the array notation. 

> I'm actually storing URIs as the keys

yeah, at one point we did something similar with assemble. We were using this lib or something similar to set templates/files on a cache, where the `key` was an absolute file path. Initially we tried some workarounds to avoid the need to escape dots. Now, we have a pretty hard and fast rule that filepaths or urls are set directly, using something like `obj[key] = val`. Mainly because it allows us to use this module without having to worry about keys being split incorrectly (I wish we had the array notation documented, it could have saved you some time).

We'll need to keep the API the same, given how much this module is used. But I'm always open to optimizations
